sql/migrations/schema-postgresql-migrate-data-to-mirrors.sql

```diff
@@ -0,0 +1,46 @@
+INSERT INTO filemetadata (
+    id,
+    path,
+    mtime,
+    size,
+    md5,
+    sha1,
+    sha256,
+    sha1piecesize,
+    sha1pieces,
+    btih,
+    pgp,
+    zblocksize,
+    zhashlens,
+    zsums
+  )
+  OVERRIDING SYSTEM VALUE
+  SELECT
+    filearr.id,
+    filearr.path,
+    to_timestamp(mtime) as mtime,
+    size::bigint,
+    md5::bytea,
+    sha1::bytea,
+    sha256::bytea,
+    sha1piecesize::integer,
+    sha1pieces::bytea,
+    btih::bytea,
+    pgp::text,
+    zblocksize::smallint,
+    zhashlens::character varying(8),
+    zsums::bytea
+  FROM filearr
+    LEFT JOIN hash ON hash.file_id = filearr.id
+  ORDER BY filearr.id;
+
+-- TODO: documentation that this needs the intarray extension
+INSERT INTO mirrors (
+    filemetadata_id,
+    server_id
+  )
+  SELECT
+    id AS filemetadata_id,
+    unnest(mirrors & (select array_agg(id) from server)) AS server_id
```

Comment:
This may need explicit type cast? I am getting 
```
/home/an/andrii-suse/environs/mb9/src/sql/migrations/schema-postgresql-migrate-data-to-mirrors.sql:46: ERROR:  operator does not exist: smallint[] & integer[]
LINE 7:     unnest(mirrors & (select array_agg(id) from server)) AS ...
                           ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
```CREATE EXTENSION intarray;```

as postgres user